### DISCUSSION TITLE
poetry run not working in heroku

### DISCUSSION THREAD

**Discussion by aahnik** [python-poetry/poetry#3994] [Q&A] [answered]
`poetry run mycommand` works in my dev environment, as well as my docker container.

deployment to heroku is being made on container stack. the heroku.yml defines

```yaml
build:
  docker:
    worker: Dockerfile

```

everything is fine.

if I pip install my package from pypi, then it runs on heroku.

but if I copy . . and the poetry install and then poetry run mycommand, it fails

something to do with the ephemeral file system of Heroku.

it says package not found.

```python
from importlib.metadata import version

__version__ = version(__package__)
```
the `__init__.py` itself fails.

is there some work around ?

i am currently doing this: (not using COPY in dockerfile, rather installing from pypi)

```Dockerfile
FROM python:3.9

WORKDIR /app

RUN apt-get update && apt-get upgrade -y

RUN pip install --upgrade pip

RUN pip install --upgrade tgcf

RUN tgcf --version && tg-login --version

CMD ["tgcf"]
```

**Comment by abn** [comment, marked as answer]
I am note sure we can help much on that one as it seems to be heroku problem. From what I can see in your message, if its building a docker image, and the source is correctly copied into the container image, there shouldn't be an issue. Things to maybe look at:

1. Ensure virtual environment is present at runtime as well as build time.
2. Poetry install was not done in a separate step that to the run command.
